.github/workflows/release.yml

```diff
@@ -3,20 +3,40 @@ on:
   push:
     branches:
       - master
+      - beta
       - "[0-9]+.[0-9]+.x"
 
+permissions:
+  contents: read # for checkout
+
 jobs:
   release:
-    name: release
+    name: Release
     runs-on: ubuntu-latest
+    permissions:
+      contents: write      # to be able to publish a GitHub release
+      issues: write        # to be able to comment on released issues
+      pull-requests: write # to be able to comment on released pull requests
+      id-token: write      # to enable use of OIDC for npm provenance
```

Comment:
All of this already worked, any reason why we're explicitly adding it? Is it because OIDC? If yes, is it enough to just add the `id-token`?

Reply:
Specifying explicit permissions helps prevent potential attacks that might try to steal the token through a compromised action or commit. In this case I've taken the permissions from the [example configuration](GitHub Actions) for GitHub Actions of the documentation.